Autonomy/app/src/main/java/nasa_rmc/autonomy/DriveState.java

```diff
@@ -0,0 +1,197 @@
+package nasa_rmc.autonomy;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by atomlinson on 3/31/17.
+ */
+
+public class DriveState implements LogicState {
+    private LogicContext logicContext;
+
+    private int motorSpeed;
+
+    private boolean isReverse;
+    public void setReverse(boolean isReverse) { this.isReverse = isReverse; }
+
+    private ArrayList<Itinerary> itineraries;
+
+    // The translation of the Tango relative to the center of the robot
+    private final double TANGO_X_TRANSLATION_FROM_CENTER = 0.0;
+    private final double TANGO_Y_TRANSLATION_FROM_CENTER = 0.0;
+
+    // The rotation of the Tango relative to the arena coordinate system
+    private final double TANGO_ACTUAL_ROTATION_ADJUSTMENT = 90.0;
+
+    // Distance and angle tolerances
+    private final double DISTANCE_TOLERANCE = 0.15;
+    private final double ANGLE_TOLERANCE = 10.0;
+
+    // Time to sleep between drive commands
+    private final int DRIVE_COMMAND_DELAY = 10;
+
+    private String status;
+    public String getStatus() { return status; }
+
+    DriveState(LogicContext logicContext, ArrayList<Itinerary> itineraries) {
+        this.logicContext = logicContext;
+        this.itineraries = itineraries;
+
+        this.motorSpeed = 25;
+        this.isReverse = false;
+    }
+
+    @Override
+    public void run() throws InterruptedException {
+        Itinerary itinerary = itineraries.remove(0);
+
+        while(!itinerary.isFinalDestinationReached()) {
+            driveToCoordinates(itinerary.getNextCoordinates());
+
+            if (hasArrivedAtCoordinates(itinerary.getNextCoordinates())) {
+                status = "At final destination.";
+                itinerary.arrivedAtCoordinates();
+            }
+
+            TimeUnit.MILLISECONDS.sleep(DRIVE_COMMAND_DELAY);
+        }
+
+        logicContext.setLogicState(logicContext.getDigState());
```

Comment:
Don't we need logic here to decide whether we transition to dig state, or dump state? I don't see how we get to the dump state currently.